xcube/core/store/descriptor.py

```diff
@@ -141,36 +169,55 @@ def __init__(self,
                          time_range=time_range,
                          time_period=time_period,
                          open_params_schema=open_params_schema)
+        self.coords = coords
         self.dims = dict(dims) if dims else None
-        self.data_vars = list(data_vars) if data_vars else None
+        if data_vars:
+            if type(data_vars) == dict:
+                self.data_vars = data_vars
+            else:
+                self.data_vars = {}
+                for data_var in data_vars:
+                    self.data_vars[data_var.name] = data_var
+        else:
+            self.data_vars = None
```

Comment:
Overall: too complex logic for simple `data_vars` assignment here. If we maintain compatibility at all, just normalize argument and assign to `self.data_vars` which is then always a dict or none.